python/src/lib.rs

```diff
@@ -85,6 +85,28 @@ struct RawDeltaTableMetaData {
     configuration: HashMap<String, Option<String>>,
 }
 
+#[inline]
+fn build_table(
+    table_uri: &str,
+    storage_options: Option<HashMap<String, String>>,
+    without_files: bool,
+    log_buffer_size: Option<usize>,
+) -> Result<DeltaTableBuilder, PythonError> {
+    let mut builder = deltalake::DeltaTableBuilder::from_uri(table_uri);
+    if let Some(storage_options) = storage_options {
+        builder = builder.with_storage_options(storage_options)
+    }
+    if without_files {
+        builder = builder.without_files()
+    }
+    if let Some(buf_size) = log_buffer_size {
+        builder = builder
+            .with_log_buffer_size(buf_size)
+            .map_err(PythonError::from)?;
+    }
+    Ok(builder)
+}
+
 #[pymethods]
 impl RawDeltaTable {
```

Comment:
I think this makes sense to add, just two comments, I think we can simplify it a lot. We can just add the load_lazy argument as input for the `new` method, something like this: 

```rust
   fn new(
        table_uri: &str,
        version: Option<i64>,
        storage_options: Option<HashMap<String, String>>,
        without_files: bool,
        log_buffer_size: Option<usize>,
        load_lazy: bool,
    ) -> PyResult<Self> {
        let options = storage_options.clone().unwrap_or_default();
        let mut builder = deltalake::DeltaTableBuilder::from_uri(table_uri);
        if let Some(storage_options) = storage_options {
            builder = builder.with_storage_options(storage_options)
        }
        if without_files {
            builder = builder.without_files()
        }
        if let Some(buf_size) = log_buffer_size {
            builder = builder
                .with_log_buffer_size(buf_size)
                .map_err(PythonError::from)?;
        }
        if let Some(version) = version {
            builder = builder.with_version(version)
        }
        let table = if !load_lazy {
            rt()?.block_on(builder.load()).map_err(PythonError::from)?
        } else {
            builder.build().map_err(PythonError::from)?
        };
        Ok(RawDeltaTable {
            _table: table,
            _config: FsConfig {
                root_url: table_uri.into(),
                options,
            },
        })
```
and in python we just do an if else to set the metadata:
```python
        if load_lazy:
            self._metadata = None
        else:
            self._metadata = Metadata(self._table)
```